bundles/org.eclipse.swt/Eclipse SWT Browser/win32/org/eclipse/swt/browser/Edge.java

```diff
@@ -486,30 +595,36 @@ public void create(Composite parent, int style) {
 
 void browserDispose(Event event) {
 	containingEnvironment.instances.remove(this);
-
-	if (webView_2 != null) webView_2.Release();
-	if (environment2 != null) environment2.Release();
-	settings.Release();
-	webView.Release();
-	webView_2 = null;
-	environment2 = null;
-	settings = null;
-	webView = null;
-
-	// Bug in WebView2. Closing the controller from an event handler results
-	// in a crash. The fix is to delay the closure with asyncExec.
-	if (inCallback) {
-		ICoreWebView2Controller controller1 = controller;
-		controller.put_IsVisible(false);
-		browser.getDisplay().asyncExec(() -> {
-			controller1.Close();
-			controller1.Release();
-		});
-	} else {
-		controller.Close();
-		controller.Release();
+	if(webViewProvider.isWebViewUnavailable()) {
```

Comment:
Since the future is not passed outside of the WebViewProvider and is not cancelled inside, how should this ever be true?

Reply:
Hmmm. I will set webViewProvider to null later in the dispose after disposing everythig and check for webViewProvider to be null for this check.